### ISSUE TITLE
Add Debug implementations for all structs

### ISSUE THREAD

**Issue by frewsxcv** [georust/geo#533]
1. Add `#![warn(missing_debug_implementations)]` to the root of each of our crates
1. Resolve all warnings

**Comment by martinfrances107** [comment]
I have a PR for this issue when handling Iterator enum .. I have resorted to outputting a indication of what is being debugged.

so the debug string looks like 

"GeometryCoordsIter::Point() Iterator"

maybe there is something better ?

**Comment by michaelkirk** [comment]
Thanks for taking a look at this @martinfrances107!

I think I'd expect something akin to [`impl Debug for Vec::Iter`](https://doc.rust-lang.org/src/core/slice/iter.rs.html#73) which includes details about the vec contents, which is more likely what you're interested in rather than just the type.

```
fn main() {
    let v = vec!["a", "b", "c"];
    println!("v.iter() debug: {:?}", v.iter());
    println!("v debug: {:?}", v);
}
```

Which outputs:
```
v.iter() debug: Iter(["a", "b", "c"])
v debug: ["a", "b", "c"]
```

**Comment by michaelkirk** [comment]
Fixed in #547
